### ISSUE TITLE
Micronaut Bean validation should support annotations on generic type argument(s)

### ISSUE THREAD

**Issue by dstepanov** [micronaut-projects/micronaut-core#4307]
2.1.0

According to BV spec 2.0 `@Valid` can be put on the generic type argument and it's preferred way:

```
public class User {

    // preferred style as of Jakarta Bean Validation 2.0
    private List<@Valid PhoneNumber> phoneNumbers;

    // traditional style; continues to be supported
    @Valid
    private List<PhoneNumber> phoneNumbers;

    // discouraged; either the container or the type argument(s) should be
    // annotated with @Valid, but not both
    @Valid
    private List<@Valid PhoneNumber> phoneNumbers;
}
```

https://beanvalidation.org/2.0/spec/#constraintdeclarationvalidationprocess-requirements-graphvalidation-examples

That doesn't work in the current version 2.1.0, validation is not applied to the collection.

**Comment by graemerocher** [comment]
Yes we need to implement the core feature to make this work which is general support for annotations on type arguments

**Comment by dstepanov** [comment]
I thought that it is supported because `AbstractBeanProperty` has `typeArguments` 🤔

**Comment by graemerocher** [comment]
Depends on #4457

**Comment by MartinKanters** [comment]
I fail to see the difference between https://github.com/micronaut-projects/micronaut-core/issues/4457 and this issue. The former one is already merged, which suggests that a property on a bean with `List<@NotBlank String> strings;` should work, but it does not seem to work on 2.5.1. Can you please let me know the status and ETA (if known) of this feature?
